### ISSUE TITLE
JS warning when changing the value of the country or state select

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#3146]
## To reproduce
You need to have Gutenberg enabled in order to reproduce this issue.

1. Go to the Checkout block.
2. Change the value of the Country/Region select.
3. In the browser console, notice there is a warning: ``Warning: Cannot update a component (`ShippingDataProvider`) while rendering a different component (`CustomSelectControl`). To locate the bad setState() call inside `CustomSelectControl`, follow the stack trace as described in https://fb.me/setstate-in-render``

## Screenshots
![Peek 2020-09-15 11-24](https://user-images.githubusercontent.com/3616980/93192582-1cbad880-f746-11ea-8228-9ddddf180f75.gif)
![imatge](https://user-images.githubusercontent.com/3616980/93192370-e5e4c280-f745-11ea-9fe9-c36b51f95d6f.png)

**Comment by haszari** [comment]
I couldn't reproduce this – I'm running WordPress 5.6-beta4 and Gutenberg 9.3.0.

@Aljullu is there any special setup needed, or perhaps this has resolved itself .. ?

##### Woo System Report
<details>

```
### WordPress Environment ###

WordPress address (URL): http://localhost:8333
Site address (URL): http://localhost:8333
WC Version: 4.8.0
REST API Version: ✔ 4.8.0
WC Blocks Version: ✔ 3.9.0-dev
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.7.0
Log Directory Writable: ✔
WP Version: 5.6-beta4
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.38 (Debian)
PHP Version: 7.4.11
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.64.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.5.8-MariaDB-1:10.5.8+maria~focal
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.8.0
WC Database Prefix: wp_
Total Database Size: 5.43MB
Database Data Size: 3.13MB
Database Index Size: 2.30MB
wp_woocommerce_sessions: Data: 0.03MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.05MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.03MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.14MB + Index: 0.17MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.72MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

customize_changeset: 3
page: 9
post: 2
product: 30
product_variation: 7
revision: 22
shop_coupon: 2
shop_order: 26

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (4) ###

Gutenberg: by Gutenberg Team – 9.3.0
WooCommerce Blocks: by Automattic – 3.9.0-dev
WooCommerce: by Automattic – 4.8.0-dev
WordPress Beta Tester: by Peter Westwood
Andy Fragen – 3.0.3


### Inactive Plugins (3) ###

Akismet Anti-Spam: by Automattic – 4.1.7
Hello Dolly: by Matt Mullenweg – 1.7.2
WooCommerce Cart Add-Ons: by WooCommerce – 1.15.32

### Must Use Plugins (2) ###

fork of Mailhog for WordPress: by Tareq Hasan – 1.0
wpcomsh-loader.php: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #5 - /?page_id=5
Cart: #6 - /?page_id=6
Checkout: ❌ Page does not contain the shortcode.
My account: #8 - /?page_id=8
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.9.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Canceled: 34
Oldest: -0001-11-30 00:00:00 +0000
Newest: -0001-11-30 00:00:00 +0000

Complete: 36
Oldest: 2020-11-08 21:31:23 +0000
Newest: 2020-11-08 21:41:28 +0000

Pending: 3
Oldest: 2020-11-03 20:21:18 +0000
Newest: 2020-11-15 21:36:56 +0000
```

</details>

**Comment by Aljullu** [comment]
That's weird. I could reproduce it with WordPress 5.6-beta4 with and without Gutenberg `master` enabled. I could also reproduce it in three browsers (Firefox, Chrome and Gnome Web - which uses the same engine as Safari).

The error appears when changing the country in the Shipping fields or the Billing fields, doesn't matter. It only happens the first time you change the country, changing it again doesn't show an error.

I will assign this to me to investigate it further.

**Comment by Aljullu** [comment]
I tested with a version of `CustomSelectControl` from Gutenberg's `master` and I couldn't reproduce the issue. So this might be fixed updating `@wordpress/components` to the latest version.

@nerrad this issue is in the current sprint but it would probably be fixed by #2140 (which is in the backlog). What do you think about re-priorizing it? Otherwise, should I just close this one in favor of #2140?

**Comment by nerrad** [comment]
We've had difficulty with updating `@wordpress/components` dependency to the latest version in the past but it's still something that's going to need to be done at some point for things we pull directly from that package as opposed to include via whatever is available in the WordPress core version we support.

Would you be able to do a timeboxed one day attempt at updating `@wordpress/components` to the latest version in our dependencies and see if that works okay.
